lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -0,0 +1,272 @@
+        fixture.
+        """
+        if not self.empty_es_response_payload or not self.elastic_endpoint:
+            pytest.skip("skipping the empty query test")
+        expected_status = HTTPStatus.OK
+        if build_auth_header["header_param"] != "valid":
+            expected_status = HTTPStatus.FORBIDDEN
+
+        index = self.build_index(server_config, ("2020-08", "2020-09", "2020-10"))
+        response = query_api(
+            self.pbench_endpoint,
+            self.elastic_endpoint,
+            self.payload,
+            index,
+            expected_status,
+            headers=build_auth_header["header"],
+            status=HTTPStatus.OK,
+            json=self.empty_es_response_payload,
+        )
+        assert response.status_code == expected_status
+        if response.status_code == HTTPStatus.OK:
+            assert response.json == []
+
+    @pytest.mark.parametrize(
+        "exceptions",
+        (
+            {
+                "exception": requests.exceptions.ConnectionError(),
+                "status": HTTPStatus.BAD_GATEWAY,
+            },
+            {
+                "exception": requests.exceptions.Timeout(),
+                "status": HTTPStatus.GATEWAY_TIMEOUT,
+            },
+            {
+                "exception": requests.exceptions.InvalidURL(),
+                "status": HTTPStatus.INTERNAL_SERVER_ERROR,
+            },
+            {"exception": Exception(), "status": HTTPStatus.INTERNAL_SERVER_ERROR},
+            {"exception": ValueError(), "status": HTTPStatus.INTERNAL_SERVER_ERROR},
+        ),
+    )
+    def test_http_exception(
+        self,
+        server_config,
+        query_api,
+        exceptions,
+        user_ok,
+        find_template,
+        pbench_token,
+    ):
+        """
+        Check that an exception in calling Elasticsearch is reported correctly.
+        """
+        if not self.elastic_endpoint:
+            pytest.skip("skipping the http exception test")
+
+        # Make the start and end time in payload the same to result in an exception
+        self.payload["end"] = self.payload["start"]
+
+        index = self.build_index(server_config, ("2020-08",))
+        query_api(
+            self.pbench_endpoint,
+            self.elastic_endpoint,
+            self.payload,
+            index,
+            exceptions["status"],
+            body=exceptions["exception"],
+            headers={"Authorization": "Bearer " + pbench_token},
+        )
+
+    @pytest.mark.parametrize("errors", (400, 500, 409))
+    def test_http_error(
+        self, server_config, query_api, user_ok, find_template, pbench_token, errors
+    ):
+        """
+        Check that an Elasticsearch error is reported correctly through the
+        response.raise_for_status() and Pbench handlers.
+        """
+        if not self.elastic_endpoint:
+            pytest.skip("skipping the http error test")
+
+        # Make the start and end time in payload the same to result in an error
```

Comment:
Again, the comment is misleading as the dates don't "result in an error" but merely ensure we can get far enough for the requests mock to generate an error in the Elasticsearch call.